eeps/eep-0064.md

```diff
@@ -0,0 +1,354 @@
+    Author: Kiko Fernandez-Reyes <kiko(at)erlang(dot)org>,
+        Raimo Niskanen <raimo(at)erlang(dot)org>
+    Status: Draft
+    Type: Standards Track
+    Created: 07-Jun-2023
+    Erlang-Version: OTP-27
+    Post-History:
+****
+EEP 64: Multi-line Indented Verbatim Strings
+----
+
+Abstract
+========
+
+This EEP proposes the introduction of Multi-line Indented Verbatim strings,
+*MIV Strings* and defines their semantics.  The main benefit is to allow
+*multi-line* strings in an easy and useful, *indented*, way,
+similar to in other languages, e.g Elixir.
+
+Their first use case is for in-module documentation attributes
+containing [Markdown][] or similarly formatted text where *verbatim* text
+is desirable since any documentation text format has its own notion
+of escape sequences which will collide with Erlang's escape sequences.
+
+Rationale
+=========
+
+Today (June 2023), writing multi-line strings is awkward and arguably ugly.
+They may contain escape sequences and have no concept of indentation:
+
+    foo () ->
+        case bar() of
+             ok ->
+                 X = "First line
+    Second line with \"\\*not emphasized\\* Markdown\"
+    Third line",
+                 {ok, X}
+        end.
+
+The content's indentation cannot adhere to the surrounding code's
+and the `*` has to be doubly escaped to get a  `\*` character
+sequence into the actual content.
+
+In a documentation attribute as suggested in [EEP 59][]
+the indentation problem is not that pronounced because
+the documentation attribute itself is not much indented:
+
+    -doc """
+        First line
+        Second line with "\*not emphasized\* Markdown"
+        Third line
+        """.
+
+The main reason to consider this EEP is for documentation
+attributes, and here not having to worry about escape sequences
+is the most attractive property of this EEP.  But introducing
+a new string format will also require defining how it would behave
+in Erlang code.
+
+Having a string format only allowed in attributes would simply be
+very strange and the one suggested in this EEP would be useful
+also in Erlang code.
+
+Design Decisions
+----------------
+
+An attribute is an Erlang form in the source code
+that consists of a `-` token, an atom, one value term and a full stop
+(dot).  The value term may be enclosed in parentheses
+(which is not very interesting for documentation attributes).
+
+    -doc "  Badly formatted
+    documentation paragraph
+    /-\\
+    \\-/".
+
+A documentation attribute should have a string as its content term,
+and here we want to use our new and more convenient MIV String
+instead of a normal string:
+
+    -doc """
+          Better formatted
+        documentation paragraph
+        /-\
+        \-/
+        """.
+
+### MIV String Scanner Token
+
+A MIV String must be a token that the scanner recognizes as a string,
```

Comment:
```suggestion
An MIV String must be a token that the scanner recognizes as a string,
```

Reply:
Since this depends on if the correct pronunciation of the abbreviation "MIV" is "emm eye vee", "miv" ("meev") or "multiline ...", which is unclear given that it is not an established abbreviation, I think I will be creative and rename the whole shebang to "Verbatim Multi-line Indented Strings", also because the EEP argues that "Verbatim" is the most urgent property.